### DISCUSSION TITLE
Feature request from Nation3: pause streams

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#14] [Ideas]
From my private chat with Luis Cuende:

<img width="547" alt="chat-1" src="https://user-images.githubusercontent.com/8782666/221109117-6e19ebf3-5529-463b-86ae-b51e79990940.png">

<img width="547" alt="chat-2" src="https://user-images.githubusercontent.com/8782666/221109128-f4fad44d-10c9-4859-a9cc-7fe8ebb18a76.png">

This could be implemented in the proxy by simply canceling the stream but not returning the funds back to the proxy owner. The proxy would "escrow" the funds when `pause` is called, and then there would be another function (e.g. `restart`), which would pick up the escrowed funds to create another stream.

**Comment by andreivladbrg** [comment]
>> but not returning the funds back to the proxy owner.

 I believe that it would require additional logic, such as a `balance` storage variable. 

The argument that the target is a contract that can be easily swapped and upgraded will no longer be valid.

I think we discussed a lot about these features (pause, restart) in the past and we agreed that they are not really worth it. Also the app is not even designed for these features. 

Therefore, I suggest that we refrain from thinking / accepting new features for V2 (proxy being the last major change) and focus on shipping it as soon as possible, given the time constraints we're facing.

**Reply by PaulRBerg** [reply]
> I believe that it would require additional logic, such as a balance storage variable.

Not necessarily. You could just check if the standard ERC-20 balance is more than the requirement needed by `restart`.

> we agreed that they are not really worth it

Yes, but now I presented you with a discussion with a potential user who has invalidated our assumptions.

> the app is not even designed for these features.

At the moment, yes. But that doesn't mean it can't be designed to support this in the future, or that non-frontend users wouldn't be interested in this feature.

> we refrain from thinking / accepting new features for V2

I disagree. We should never stop talking to users.

As long as the core can remain intact, there's no harm in innovating in the periphery.

> given the time constraints we're facing

For the avoidance of doubt - I'm not suggesting we implement this feature right away.

**Reply by andreivladbrg** [reply]
> Not necessarily. You could just check if the standard ERC-20 balance is more than the requirement needed by restart.

I am not going into technical details now.

> I disagree. We should never stop talking to users.

Ofc we should never do that, my point was that until we launch V2 as is designed at the moment to not make promises and just give a kind answer such as "Interesting idea, we will consider it after we ship the current product".

> For the avoidance of doubt - I'm not suggesting we implement this feature right away.

Ok, let's lock this discussion for the time being and talk about it in the context of V2.1.

**Comment by smol-ninja** [comment]
Since Nation3 requested for `pause` feature, would they be a potential user for `Flow` protocol as we have `pause` and `restart` there?

**Reply by PaulRBerg** [reply]
I don't think they exist anymore. cc @maxdesalle.

**Reply by maxdesalle** [reply]
Even though Luis left, Nation3 isn't dead @PaulRBerg AFAIK, you can see they still have activity here: https://github.com/nation3

I also remember seeing an Airstream campaign from them quite recently.

**Reply by maxdesalle** [reply]
We could definitely ask Luis to put us in touch with the current team @smol-ninja when we launch Flow to pitch them the product.

**Reply by PaulRBerg** [reply]
Oh, I didn't know that. Let's add them to the CRM then.

**Comment by PaulRBerg** [comment]
Closing this as Flow supports pausing streams.
